util/compression_test.cc

```diff
@@ -220,6 +220,161 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
+              "SetCompressionTimeOutputSize",
+              set_compress_time_and_size(1000, 100));
+          break;
+        case 1:
+          // Verify that the Mocked cpu cost and io cost are predicted correctly
+          EXPECT_EQ(predicted_io_bytes_, 100);
+          EXPECT_EQ(predicted_cpu_time_, 1000);
+          SyncPoint::GetInstance()->SetCallBack(
+              "CostAwareCompressor::CompressBlock::"
+              "SelectCompressionTypeAndLevel",
+              set_compression_type_and_level(6, 2));
+          SyncPoint::GetInstance()->SetCallBack(
+              "CostAwareCompressor::CompressBlockAndRecord::"
+              "SetCompressionTimeOutputSize",
+              set_compress_time_and_size(1000, 100));
+          break;
+      }
+      SyncPoint::GetInstance()->EnableProcessing();
+    }
+    num_keys_++;
+    return true;
+  }
+
+ private:
+  int window_;
+  int num_keys_;
+  const BlockBuilder& data_block_builder_;
+
+  static size_t cpu_time_, predicted_cpu_time_, predicted_io_bytes_;
+  static size_t output_size_;
+};
+size_t CostAwareTestFlushBlockPolicy::cpu_time_ = 0,
+       CostAwareTestFlushBlockPolicy::predicted_cpu_time_ = 0,
+       CostAwareTestFlushBlockPolicy::predicted_io_bytes_ = 0,
+       CostAwareTestFlushBlockPolicy::output_size_ = 0;
+class CostAwareTestFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
+ public:
+  explicit CostAwareTestFlushBlockPolicyFactory(const int window)
+      : window_(window) {}
+
+  virtual const char* Name() const override {
+    return "CostAwareTestFlushBlockPolicyFactory";
+  }
+
+  virtual FlushBlockPolicy* NewFlushBlockPolicy(
+      const BlockBasedTableOptions& /*table_options*/,
+      const BlockBuilder& data_block_builder) const override {
+    (void)data_block_builder;
+    return new CostAwareTestFlushBlockPolicy(window_, data_block_builder);
+  }
+
+ private:
+  int window_;
+};
+class DBCompresssionCostPredictor : public DBTestBase {
+ public:
+  Options options;
+  DBCompresssionCostPredictor()
+      : DBTestBase("db_cpuio_skip", /*env_do_fsync=*/true),
+        options(CurrentOptions()) {
+    options.compression_manager = CreateCostAwareCompressionManager();
+    if (ZSTD_Supported()) {
+      options.compression = kZSTD;
+      options.bottommost_compression = kZSTD;
+    }
+    auto statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
+    options.statistics = statistics;
+    options.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
+    BlockBasedTableOptions bbto;
+    bbto.enable_index_compression = false;
+    bbto.flush_block_policy_factory.reset(
+        new CostAwareTestFlushBlockPolicyFactory(10));
+    options.table_factory.reset(NewBlockBasedTableFactory(bbto));
+    DestroyAndReopen(options);
+  }
+};
+TEST_F(DBCompresssionCostPredictor, CostAwareCompressorManager) {
+  // making sure that the compression is supported
+  if (ZSTD_Supported()) {
+    const int kValueSize = 20000;
+    auto index_ = 0;
```

Comment:
Please don't use `auto something = <number literal>;' because it's not clear what type is going to be assigned to it, making it hard to predict when we'll get compiler warnings (as errors) about implicit narrowing or signed/unsigned mismatch.

Nit: trailing '_' only for member variables.

And 'index' is a confusing name considering its many meanings. How about 'next_key'?